### ISSUE TITLE
File watching at Linux not working

### ISSUE THREAD

**Issue by sphh** [aardappel/treesheets#148]
The file watching feature works on Windows 10 (but not on 7, as I recently found out). But it is also not working on Linux (installed with the latest AppImage. I read somewhere in the discussions of TreeSheet (https://groups.google.com/forum/#!topic/treesheets/PyozLnrpOjk), that this is a "defect" of the Linux version of wxWidget and that you want to work on it. Has that been resolved?

**Comment by aardappel** [comment]
No, that hasn't been fixed yet. The code makes use of wxWidgets functionality that has apparently never been fixed: https://github.com/aardappel/treesheets/blob/327bf29d0232ccc3ee0f9e3e4686ff322b31c441/src/myframe.h#L1080-L1130

I have a todo item about it: https://github.com/aardappel/treesheets/blob/327bf29d0232ccc3ee0f9e3e4686ff322b31c441/TODO.txt#L35-L36

I think its easier to replace this convoluted code by some code that checks the file date every 10 seconds or so, that should work equally on all platforms, is simpler, and shouldn't be much of a performance burden.

**Comment by sphh** [comment]
What about using Linux's inotify native kernel service?

Google thinks, that MacOS and Windows have similar features. [fswatch](https://github.com/emcrisostomo/fswatch) implements this for a lot of operating system. Just an idea ...

**Comment by aardappel** [comment]
Yup, I am sure it can be solved that way too. But I am unlikely to do a bunch of OS-specific programming, or pull in a large library to do the same. I like to keep things simple/portable.

**Comment by sphh** [comment]
No worries, that was just an idea ... ;-) Would love to have this sorted!

**Comment by sphh** [comment]
So sorry, that this is not planned, since this is one of the great features of TreeSheets: Many people working on one file. Sad that you outruled alternative solutions, such as `libfswatch` (or others).

**Comment by tobiolo** [comment]
I think that issue has been solved. Maybe some container technology blocks this on Linux but we cannot cover every corner case on a specific platform.
